scalding-parquet-scrooge/src/main/scala/com/twitter/scalding/parquet/scrooge/ParquetCollectionFormatForwardCompatibility.scala

```diff
@@ -0,0 +1,153 @@
+package com.twitter.scalding.parquet.scrooge
+
+import java.util
+
+import org.apache.parquet.schema.Type.Repetition
+import org.apache.parquet.schema.{GroupType, MessageType, Type}
+import org.apache.parquet.thrift.DecodingSchemaMismatchException
+import org.slf4j.LoggerFactory
+
+import scala.collection.mutable
+
+object ParquetCollectionFormatForwardCompatibility {
+
+  private val LOGGER = LoggerFactory.getLogger(getClass)
+
+  /**
+   * Create a forward-compatible schema, using content from source type with format from target type.
+   * @param sourceType source type with legacy format
+   * @param targetType target type to which source is converted to
+   */
+  def formatForwardCompatibleMessage(sourceType: MessageType, targetType: MessageType): MessageType = {
+    val groupResult = formatForwardCompatibleType(sourceType, targetType).asGroupType()
+    LOGGER.info("Making source schema to be compatible with target" +
```

Comment:
Should this be `info` level by default? Or will that be too verbose? I took a look at the existing code to try to find precedent but couldn't identify an obvious existing convention.

Reply:
I think too verbose, the schemas can often be very large. Maybe debug level?

Reply:
Updated to debug.